.github/workflows/push-broker.yaml

```diff
@@ -0,0 +1,37 @@
+---
+jobs:
+  build-push:
+    runs-on: 'ubuntu-latest'
+    steps:
+    - uses: 'actions/checkout@v4'
+    - id: 'metadata'
+      uses: 'docker/metadata-action@v5'
+      with:
+        images: 'econialabs/emojicoin-broker'
+        tags: |
+          type=ref,event=branch
+          type=semver,pattern={{version}}
+    - uses: 'docker/setup-qemu-action@v3'
+    - uses: 'docker/setup-buildx-action@v3'
+    - uses: 'docker/login-action@v3'
+      with:
+        password: '${{ secrets.DOCKERHUB_TOKEN }}'
+        username: '${{ secrets.DOCKERHUB_USERNAME }}'
+    - uses: 'docker/build-push-action@v6'
+      with:
+        cache-from: 'type=gha'
+        cache-to: 'type=gha,mode=max'
+        context: 'src/indexer'
+        file: 'src/indexer/broker/Dockerfile'
+        labels: '${{ steps.metadata.outputs.labels }}'
+        platforms: '${{ vars.DOCKER_IMAGE_PLATFORMS }}'
+        push: 'true'
+        tags: '${{ steps.metadata.outputs.tags }}'
+name: 'Build broker Docker image and push to Docker Hub'
+'on':
+  push:
+    branches:
+    - 'unstable'
+    tags:
+    - 'v*'
```

Comment:
Should specify broker in tag

Reply:
I'm not sure. A tag is for a version that contains a working broker, processor, api and frontend. If we start tagging each component individually, how would we deploy on AWS ? It's easier to say depoly `v1.5.2` rather than deploy `broker-v1.2.3` and `processor-v2.1.0` and `frontend-v1.4.2` and then realize that those are actually not compatible and you need some other version. It's like a DSS version: we don't have a version for the aggregator, one for the processor, one for MQTT, etc., we have a version of the stack. Here it's the same thing IMO.

Reply:
> If we start tagging each component individually, how would we deploy on AWS?

The stack deployment file simply specifies the image tag for the binaries inside, like a cargo crate specifies versions of each internal dependency. Segmenting versions by service will also help with faster build times and debugging updates to individual services: https://github.com/econia-labs/emojicoin-dot-fun/blob/5557744537021842147eff0be955448c6580e0de/src/indexer/cloud-formation/deploy-dev.yaml#L6

> we have a version of the stack

Yes, this is fine, and will work with the above: once something gets merged to `main` (the testing branch) and works, we merge `main` to `indexer-stable` and issue a tag to it just like on the DSS

Reply:
I think specifying the version in a file is not a good idea. I think it is simpler to have a global version.

AWS would have two deployments:

- The latest global tag.
- Whatever is currently on main.

This way, we can easily track our stable deployment and our edge deployment.

How would we do this with versioning inside a file ?

Also, if we ever need to roll back, it will be close to impossible to know which version of the processor works with which version of the broker.

Since these services are closely linked together and evolve together, it only makes sense to have one version, IMO.